### ISSUE TITLE
simdiag does not always return orthonormal eigenvectors

### ISSUE THREAD

**Issue by Sola85** [qutip/qutip#2268]
### Bug Description

In some rare cases, like the example below, simdiag does not return orthonormal eigenvectors, despite this always being possible.

### Code to Reproduce the Bug

```shell
import numpy as np
from qutip import simdiag, Qobj

a = np.array([[1,  0, 1, -1, 0],
              [0,  4, 0,  0, 1],
              [1,  0, 4,  1, 0],
              [-1, 0, 1,  4, 0],
              [0,  1, 0,  0, 4]])

b = np.eye(5)

evals, evecs = simdiag([Qobj(a), Qobj(b)])
evecs = np.array([evec.full() for evec in evecs]).squeeze()

# Check that eigenvectors form an othonormal basis (<=> matrix of eigenvectors is unitary)
deviation_from_unitarity = np.linalg.norm(evecs@evecs.conj().T - np.eye(len(evecs)))
assert np.isclose(deviation_from_unitarity, 0)
```

### Code Output

```shell
AssertionError
```

### Expected Behaviour

Since by requirement of simdiag all input matrices must be hermitian and commute, the common eigenvectors should always be orthonormal.

### Your Environment

```shell
QuTiP Version:      4.7.1
Numpy Version:      1.23.5
Scipy Version:      1.10.1
Cython Version:     None
Matplotlib Version: 3.7.1
Python Version:     3.11.6
Number of CPUs:     8
BLAS Info:          OPENBLAS
OPENMP Installed:   False
INTEL MKL Ext:      False
Platform Info:      Linux (x86_64)
```

### Additional Context

The fix is simple: 
Here
```python
subspace = vecs.conj().T @ ops[i].full() @ vecs
eigvals, eigvecs = la.eig(subspace)
```
`la.eig` has to be replaced by `la.eigh`

**Comment by hodgestar** [comment]
Thanks @Sola85. Would you mind trying this on 4.7.3?

**Comment by Sola85** [comment]
> Thanks @Sola85. Would you mind trying this on 4.7.3?

Yes, this is reproducable in 4.7.3 and in the current master branch

**Comment by Sola85** [comment]
Fixed in PR #2269
